src/sync/sync_permission.cpp

```diff
@@ -170,16 +171,22 @@ void Permissions::set_permission(std::shared_ptr<SyncUser> user,
                                  PermissionChangeCallback callback,
                                  const ConfigMaker& make_config)
 {
+    using namespace std::chrono;
     const auto realm_url = user->server_url() + permission.path;
     auto realm = Permissions::management_realm(std::move(user), make_config);
     CppContext context;
 
+    // Get the current time.
+    int64_t ns_since_epoch = duration_cast<nanoseconds>(system_clock::now().time_since_epoch()).count();
```

Comment:
This code assumes that the system clock's epoch is the UNIX epoch that `Timestamp` expects. I don't think there's any guarantee that this is the case.